modules/basis_universal/image_compress_basisu.cpp

```diff
@@ -212,15 +210,62 @@ Ref<Image> basis_universal_unpacker_ptr(const uint8_t *p_data, int p_size) {
 	// Get supported compression formats.
 	bool bptc_supported = RS::get_singleton()->has_os_feature("bptc");
 	bool astc_supported = RS::get_singleton()->has_os_feature("astc");
+	bool rgtc_supported = RS::get_singleton()->has_os_feature("rgtc");
 	bool s3tc_supported = RS::get_singleton()->has_os_feature("s3tc");
 	bool etc2_supported = RS::get_singleton()->has_os_feature("etc2");
 
 	bool needs_ra_rg_swap = false;
 
 	switch (*(uint32_t *)(src_ptr)) {
+		case BASIS_DECOMPRESS_R: {
+			if (rgtc_supported) {
+				basisu_format = basist::transcoder_texture_format::cTFBC4_R;
+				image_format = Image::FORMAT_RGTC_R;
+			} else if (s3tc_supported) {
+				basisu_format = basist::transcoder_texture_format::cTFBC1;
+				image_format = Image::FORMAT_DXT1;
+			} else if (etc2_supported) {
+				basisu_format = basist::transcoder_texture_format::cTFETC2_EAC_R11;
+				image_format = Image::FORMAT_ETC2_R11;
+			} else {
+				// No supported VRAM compression formats, decompress.
+				basisu_format = basist::transcoder_texture_format::cTFRGBA32;
+				image_format = Image::FORMAT_RGBA8;
+			}
+
+		} break;
 		case BASIS_DECOMPRESS_RG: {
-			// RGTC transcoding is currently performed with RG_AS_RA, fail.
-			ERR_FAIL_V(image);
+			if (rgtc_supported) {
+				basisu_format = basist::transcoder_texture_format::cTFBC5_RG;
+				image_format = Image::FORMAT_RGTC_RG;
+			} else if (s3tc_supported) {
+				basisu_format = basist::transcoder_texture_format::cTFBC3;
+				image_format = Image::FORMAT_DXT5_RA_AS_RG;
+			} else if (etc2_supported) {
+				basisu_format = basist::transcoder_texture_format::cTFETC2_EAC_RG11;
+				image_format = Image::FORMAT_ETC2_RG11;
+			} else {
+				// No supported VRAM compression formats, decompress.
+				basisu_format = basist::transcoder_texture_format::cTFRGBA32;
+				image_format = Image::FORMAT_RGBA8;
+				needs_ra_rg_swap = true;
+			}
+
+		} break;
+		case BASIS_DECOMPRESS_RG_AS_RA: {
+			if (s3tc_supported) {
+				basisu_format = basist::transcoder_texture_format::cTFBC3;
+				image_format = Image::FORMAT_DXT5_RA_AS_RG;
+			} else if (etc2_supported) {
+				basisu_format = basist::transcoder_texture_format::cTFETC2;
+				image_format = Image::FORMAT_ETC2_RA_AS_RG;
+			} else {
+				// No supported VRAM compression formats, decompress.
+				basisu_format = basist::transcoder_texture_format::cTFRGBA32;
+				image_format = Image::FORMAT_RGBA8;
```

Comment:
Couldn't `Image::FORMAT_RG8` or `Image::FORMAT_LA8` be used here?

Reply:
BasisU doesn't allow directly decompressing textures to R8/RG8, so they have to be converted after decompression. This increases the loading time slightly, though I think the tradeoff is worth it since less VRAM is wasted.